### ISSUE TITLE
No applicable renderer found.

### ISSUE THREAD

**Issue by Mehdi40** [eclipsesource/jsonforms#1411]
Hello,

First of all, thanks for you work. JSONForms seems to be a really complete and documented tools.

**Describe the bug**
I encoutered a bug when trying to implement JSONForms in our codebase (React/Redux). Our architecture being what it is, I manage to implement this tool (well, that's what I thought).
In fact, what's happening, is simply that materialRenderers and materialCells, in the store, become array of empty objects.
A simple console.log shows me that, before being pushed to the store, both the objects are what they are expected to be (array of objects with tester/renderer functions).
Thus driving me to a "No applicable renderer found." error when trying to display the form.

**To Reproduce**

**Screenshots**
Console.logs:
![Capture du 2019-06-20 12-31-53](https://user-images.githubusercontent.com/19344181/59842760-753ed280-9357-11e9-8e30-ebb6df2c4edf.png)

The store (at least, the part we're interested in):
![Capture du 2019-06-20 12-33-35](https://user-images.githubusercontent.com/19344181/59842873-b46d2380-9357-11e9-96c0-479202cae365.png)

The action (INIT) being displayed with example data (person):
![Capture du 2019-06-20 12-33-49](https://user-images.githubusercontent.com/19344181/59842906-c8b12080-9357-11e9-9914-964c689c6d8d.png)

**Browser (please complete the following information):**
 - Browser [chrome]

**Used Setup (please complete the following information):**
 - Framework: [react, redux, nodejs]
 - RendererSet: [tried with material, vanilla]

Thanks in advance, if you think any more information could help, don't hesitate.

**Comment by edgarmueller** [comment]
Thanks for the report! I'm not yet sure how you interact with JSON Forms: when do the renderers become an empty array? It's not yet clear to me when reading your description. Ideally, can you provide a reproducer?

**Comment by Mehdi40** [comment]
Thanks for your quick answer!
After reading again what I've typed, I realized that this was not really clear.

Well, long story short, this was not a bug at all, this was just me not respecting the tutorials steps because of our huge codebase.
I did not used the `combineReducer()` method in the store, I just tried to set a `jsonforms` key with the JSONForms Reducer.
After reworking our entire store, we managed to make it work.

TL;DR: The problem came from me, not using JSONForms like it was designed to.

**Comment by edgarmueller** [comment]
Glad you figured it out, I'm closing this issue therefore. If there's anything let us know!
